code/messaging/src/main/java/com/adobe/marketing/mobile/aepcomposeui/utils/UIAction.kt

```diff
@@ -11,9 +11,15 @@
 
 package com.adobe.marketing.mobile.aepcomposeui.utils
 
-class UIAction {
-    companion object {
-        const val CLICK = "click"
-        const val EXPAND = "expand"
+enum class UIAction(private val action: String) {
+    CLICK("click"),
+    DOUBLE_CLICK("doubleClick"),
+    DRAG("drag"),
+    EXPAND("expand"),
```

Comment:
- `Expand` is a special type of click (similar to dismiss) and the result of the click is an expand. IMO this should not be a separate action. 
- Let us only add actions that we need for now (only clicks I think for now)
- If this is also made a sealed class, we can let each action have its own set of properties while allowing consumers to use `when` to identify the action.

sealed class UiAction(id: String) {
    data class Click(val id: String, val actionUrl: String) : UiEvent(id)  // Represents a click with some action
   ....
}